lib/libvarnish/vcli_serve.c

```diff
@@ -428,6 +440,12 @@ VCLS_AddFd(struct VCLS *cs, int fdi, int fdo, cls_cb_f *closefunc, void *priv)
 	cfd->cli->vlu = VLU_New(cfd, cls_vlu, *cs->maxlen);
 	cfd->cli->sb = VSB_new_auto();
 	cfd->cli->limit = cs->limit;
+	if (cs->ws_sz > 0) {
+		cfd->cli->ws = malloc(sizeof(*cfd->cli->ws));
```

Comment:
this is the malloc for the struct ws itself. the malloc for the workspace happens in line 447
reading the cache_param here would only be possible if the code lived in cache, but it doesn't. the code here need to work with both cache and mgt.